activerecord/test/cases/adapter_test.rb

```diff
@@ -261,6 +283,14 @@ def test_log_invalid_encoding
         assert_not_nil error.message
       end
     end
+
+    private
+
+      def syntax_error_exception_class
+        return Mysql2::Error if defined?(Mysql2)
+        return PG::SyntaxError if defined?(PG)
+        return SQLite3::SQLException if defined?(SQLite3)
```

Comment:
This covers the three in-tree adapters, but it feels brittle to return `nil` by default here. I know that the [`oracle-enhanced` adapter](https://github.com/rsim/oracle-enhanced) runs the Active Record test suite to verify their implementation; this test will break for them.

Do we need to verify the class of the wrapped exception? Is it realistically possible for the assertion to fail (since Ruby is setting the exception's cause automatically)?